sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileClient.java

```diff
@@ -954,9 +967,53 @@ public Response<ShareFileProperties> downloadToFileWithResponse(String downloadF
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<ShareFileProperties> downloadToFileWithResponse(String downloadFilePath, ShareFileRange range,
         ShareRequestConditions requestConditions, Duration timeout, Context context) {
-        Mono<Response<ShareFileProperties>> response = shareFileAsyncClient.downloadToFileWithResponse(downloadFilePath,
-            range, requestConditions, context);
-        return StorageImplUtils.blockWithOptionalTimeout(response, timeout);
+        Objects.requireNonNull(downloadFilePath, "'downloadFilePath' cannot be null");
+        ShareRequestConditions finalRequestConditions = requestConditions == null ? new ShareRequestConditions() : requestConditions;
+
+        Callable<Response<ShareFileProperties>> operation = () -> {
+            try (FileChannel fileChannel = FileChannel.open(Paths.get(downloadFilePath), StandardOpenOption.WRITE, StandardOpenOption.CREATE_NEW)) {
```

Comment:
The new implementation for this method is missing the partial download retry mechanism that is available in the asynchronous download method. The asynchronous method contains:

```java
.flatMap(fbb -> FluxUtil
    .writeFile(fbb, channel, chunk.getStart() - (range == null ? 0 : range.getStart()))
    .retryWhen(Retry.max(3).filter(throwable -> throwable instanceof IOException
        || throwable instanceof TimeoutException))))
```

Which will retry downloading a specific chunk three times if there is a failure during writing to the file after the initial response has complete. In other terms, if the service returns a non-error status code and if writing to file fails due to an error in file writing or reading the bytes from the response it will retry the request.

Reply:
This should be handled now at line 1086,
`if (retries > 3) {`